toke.c

```diff
@@ -8713,7 +8713,7 @@ yyl_try(pTHX_ char *s)
 	    }
 	    if (PL_minus_E)
 		sv_catpvs(PL_linestr,
-			  "use feature ':5." STRINGIFY(PERL_VERSION) "';");
+			  "use feature ':" STRINGIFY(PERL_VERSION_MAJOR) "." STRINGIFY(PERL_VERSION_MINOR) "';");
```

Comment:
FYI: Wow, I had no idea that `-E` was implemented in quite this way. We may want to look into that and neaten it up as part of the wider look at -e/-E  ;)

Reply:
yeap this is also not optimal as this trigger the load of feature.pm
in my initial experimental branch `p7` I played quite a lot with this
I was also playing with the idea of `-5`, `-7` switches... but this is a discussion for another day